config/application.rb

```diff
@@ -38,6 +38,7 @@ class Application < Rails::Application
     config.i18n.enforce_available_locales = true
 
     config.active_record.observers = :liquid_partial_observer
+    config.assets.paths += (Settings.custom_assets_paths || "").split(":") if Settings.custom_assets_paths.present?
 
```

Comment:
I think the ```split``` and nil check and so on should go into the ```Settings``` class. It's the one knowing about the format of ```custom_assets_paths``` and this knowledge should not be required by callers. So IMHO ```Settings.custom_assets_path``` should always return an Array of Strings. Or an empty Array. Then one can safely concatenate with Rails config. And it's possible to write a test.